scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -383,19 +383,41 @@ class FossId(
         return results
     }
 
+    private suspend fun List<Scan>.findLatestPendingOrFinishedScan(url: String, revision: String? = null): Scan? {
+        return sortedByDescending { scan -> scan.id }.find { scan ->
+            // The scans in the server contain the url with the credentials so we have to remove it for the
+            // comparison. If we don't, the scans won't be matched if the password changes!
+            val urlWithoutCredentials = scan.gitRepoUrl?.replaceCredentialsInUri()
+            if (urlWithoutCredentials == url && (revision == null || (scan.gitBranch == revision))) {
```

Comment:
Maybe you could extract the `if` condition to an additional `filter()` call executed before `sortedByDescending`. This would simplify the lambda passed to `find` and would be slightly more efficient because the list is reduced before it is sorted.